src/app/app.component.ts

```diff
@@ -28,31 +32,40 @@ import { selectBaseCurrency } from './store/trips/selectors';
     MatToolbarModule,
     MatIconModule,
     MatButtonModule,
+    AsyncPipe,
   ],
   templateUrl: './app.component.html',
   styleUrl: './app.component.scss',
 })
 export class AppComponent {
   title = 'bon-voyage';
+  isLoggedIn$ = this.userStore.select(selectIsLoggedIn);
 
   constructor(
-    private store: Store<TripState>,
-    private router: Router,
+    private tripStore: Store<TripState>,
+    private userStore: Store<UserState>,
     protected authService: AuthService
   ) {
-    this.store
-      .select(selectBaseCurrency)
-      .pipe(takeUntilDestroyed())
+    this.isLoggedIn$
+      .pipe(
+        takeUntilDestroyed(),
+        switchMap(isLoggedIn => {
+          if (isLoggedIn) {
+            return this.userStore.select(selectBaseCurrency);
+          } else {
+            return EMPTY;
+          }
+        })
+      )
       .subscribe(baseCurrency => {
-        this.store.dispatch(
-          getExchangeRates({ baseCurrency: baseCurrency ?? 'ZAR' })
-        );
+        if (baseCurrency) {
+          this.tripStore.dispatch(getExchangeRates({ baseCurrency }));
```

Comment:
Consider dispatching this action within the effects so immediately after the user is logged in, the exchange rates get be retrieved for that user. This will leave the component with only the selector (less responsibilities)

Reply:
As discussed, resolved